gemstash.gemspec

```diff
@@ -32,6 +32,7 @@ you push your own private gems as well."
   spec.required_ruby_version = ">= 2.4"
 
   spec.add_runtime_dependency "activesupport", ">= 4.2", "< 6"
+  spec.add_runtime_dependency "aws-sdk-s3", "~> 1.67"
```

Comment:
Let's make this dependency work the same way as redis or mysql down on lines 45-50, where it is not a hard requirement. Instead, document that anyone wanting to use gemstash with S3 will need to install the aws-sdk-s3 gem as wall. That way gemstash users who don't want to use s3 don't have to install the s3 gem.

Reply:
Sorry for the late reply, but if I'm understanding this correctly, if this is added, the user will need to manually edit gemstash's gemspec to include the s3 gem in-order for this feature to work right? I'm just making sure it is added as intended.

Reply:
@duyquangnguyenhac nope. The user does't have to do that they will just need to install the gem manually.

Add it just as a development dependency, just like redis.

Reply:
Yeah, no editing of gemspec. Either they run “gem install”, or they use a Gemfile that has both gemstash and aws-sdk-s3 in it.

If you want to get fancy, you can rescue errors while trying to require and print a message that the user needs to install aws-sdk-s3 and try again. 🙂

Reply:
Done 👍
Moved the gem into development dependency and adds a custom error message when requiring load error occurs.
Added in d688581.